akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala

```diff
@@ -8,22 +8,24 @@ import scala.annotation.tailrec
 
 import akka.actor.typed.scaladsl.Behaviors
 
+/**
+ * A message protocol for actors that support spawning a child actor when receiving a [[SpawnProtocol#Spawn]]
+ * message and sending back the [[ActorRef]] of the child actor. An implementation of a behavior for this
+ * protocol is defined in [[SpawnProtocol#behavior]]. That can be used as is or composed with other behavior
+ * using [[Behavior#orElse]].
+ *
+ * The typical usage of this is to use it as the guardian actor of the [[ActorSystem]], possibly combined with
+ * `Behaviors.setup` to starts some initial tasks or actors. Child actors can then be started from the outside
+ * by telling or asking [[SpawnProtocol#Spawn]] to the actor reference of the system. When using `ask` this is
+ * similar to how [[akka.actor.ActorSystem#actorOf]] can be used in untyped actors with the difference that
+ * a `Future` / `CompletionStage` of the `ActorRef` is returned.
+ *
+ * Stopping children is done through specific support in the protocol of the children, or stopping the entire
+ * spawn protocol actor.
+ */
```

Comment:
I know you just moved it, but I think we should update this documentation, as it's no longer accurate.

Reply:
You mean the or-else part? I think the rest is correct (and the stopping part I just added since I saw questions on discuss about that).

Reply:
yeah `orElse` and it mentions `.behavior` rather than `.apply` still

Reply:
Fixed!